drivers/platform/stm32/stm32_spi.c

```diff
@@ -283,15 +281,42 @@ int32_t stm32_spi_transfer(struct no_os_spi_desc *desc,
 	}
 
 	for (uint32_t i = 0; i < len; i++) {
-		rx_cnt = 0;
-		tx_cnt = 0;
+
+		if (!msgs[i].tx_buff && !msgs[i].rx_buff)
+			return -EINVAL;
 
 		/* Assert CS */
 		gdesc->port->BSRR = NO_OS_BIT(sdesc->chip_select->number) << 16;
 
 		if(msgs[i].cs_delay_first)
 			no_os_udelay(msgs[i].cs_delay_first);
 
+#ifndef SPI_SR_TXE
+		/* Some STM32 families have different naming for
+		   SPI_SR_TXE, SPI_SR_RXNE and SPIx->DR. In that case, simply
+		   use the HAL API for SPI transmission, which is generic
+		   for all STM32 families. */
+
+		if (msgs[i].tx_buff && msgs[i].rx_buff)
+			ret = HAL_SPI_TransmitReceive(&sdesc->hspi, msgs[i].tx_buff, msgs[i].rx_buff,
```

Comment:
In order to simplify things, could HAL_SPI_TransmitReceive do the job all by itself here ? I guess it depends if it can handle NULL pointers for tx and rx buffers.

Reply:
Unfortunately it cannot handle null pointers, here is an example for $3 family: https://github.com/cnoviello/mastering-stm32/blob/master/nucleo-f411RE/system/src/stm32f4xx/stm32f4xx_hal_spi.c#L840